src/Database/LSMTree/Internal/Run/Mutable.hs

```diff
@@ -216,3 +219,34 @@ writeIndexFinalChunk _ _ _ = return ()
 -- Does not write anything, serialisation for the index is not yet implemented.
 writeIndexChunks :: HasFS IO h -> MRun (FS.Handle h) -> [Index.Chunk] -> IO ()
 writeIndexChunks _ _ _ = return ()
+
+-- Does not write anything, serialisation for the filter is not yet implemented.
+writeFilter :: HasFS IO h -> MRun (FS.Handle h) -> Bloom SerialisedKey -> IO ()
+writeFilter _ _ _ = return ()
+
+{-------------------------------------------------------------------------------
+  ChecksumHandle
+-------------------------------------------------------------------------------}
+
+-- | Tracks the checksum of a (write mode) file handle.
+data ChecksumHandle fhandle = ChecksumHandle !fhandle !(IORef CRC32C)
```

Comment:
Maybe note here too that this isn't meant to be concurrently accessed

Reply:
I think it's fine, since this is only used internally within this module and the `MRun` itself has a note that it can only be used in a single-threaded manner.